packages/pipeline/migrations/1543434472116-TokenOrderbookSnapshots.ts

```diff
@@ -0,0 +1,33 @@
+import { MigrationInterface, QueryRunner, Table } from 'typeorm';
+
+const tokenOrderbookSnapshots = new Table({
+    name: 'raw.token_orderbook_snapshots',
+    columns: [
+        { name: 'source', type: 'varchar', isPrimary: true },
```

Comment:
Maybe I'm missing something but are all of these columns supposed to be primary columns? At the very least I would think that `base_volume` and `quote_volume` should not be primary columns.

Reply:
Hm. This is my same thought. I was under the impression that primary key(s) have to uniquely identify each row. In this case, I think I need all of these columns to uniquely identify a row! The volumes are necessary because for each market, there is one row per price point and the price point is jointly specified by the volumes.

Reply:
Uniqueness could be in time / base symbol / quote symbol / price. I see you have price data below, can you keep it?

I guess theres no way to avoid having a floating point index if we want ordering in the physical storage, but can we somehow ensure time is the first index when sorting for performance reasons?

Reply:
Agree with @wakkadojo. For this level of abstraction, I think it makes sense to store price and use it instead of the volumes as the primary key.

This will also line up better with the types of queries we will run, which will often involve sorting by price.